great_expectations/data_context/data_context.py

```diff
@@ -3259,14 +3266,107 @@ def test_yaml_config(
         }
 
         usage_stats_event_name: str = "data_context.test_yaml_config"
+
+        config = self._test_yaml_config_prepare_config(
+            yaml_config, runtime_environment, usage_stats_event_name
+        )
+
+        if "class_name" in config:
+            class_name = config["class_name"]
+
+        instantiated_class: Any = None
         usage_stats_event_payload: Dict[str, Union[str, List[str]]] = {}
 
         if pretty_print:
             print("Attempting to instantiate class from config...")
+        try:
+            if class_name in self.TEST_YAML_CONFIG_SUPPORTED_STORE_TYPES:
+                (
+                    instantiated_class,
+                    usage_stats_event_payload,
+                ) = self._test_instantiation_of_store_from_yaml_config(
+                    name, class_name, config
+                )
```

Comment:
The majority of the refactor is here. Since we split up the logic by class type, I've created separate helpers to cover each type. This will be extended to include RBP down the line.